server/plugin.go

```diff
@@ -50,13 +47,13 @@ type Plugin struct {
 
 	// configuration and a muttex to control concurrent access
 	conf     config
-	confLock sync.RWMutex
+	confLock *sync.RWMutex
```

Comment:
In case a method is defined by value, as in `func (p Plugin) someMethod` you get a `copy lock by value` warning (at least in some go vet configurations). I don't think it really makes sense to define methods with a value receiver on `Plugin`, but didn't want to preclude the possibility.

Reply:
I would rather leave this standard, since it's the way it is in mattermost-plugin-sample. Then just avoid value receivers.

Reply:
NP. I agree, consistency is important. Will change.

> Then just avoid value receivers.

This is not just for us, for all plugin developers, right?

Reply:
Correct. For all plugin developers.

Reply:
done